core/creator/src/main/java/io/quarkus/creator/phase/runnerjar/RunnerJarPhase.java

```diff
@@ -346,6 +340,52 @@ public void accept(Path path) {
         }
     }
 
+    /**
+     * Manifest generation is quite simple : we just have to push some attributes in manfest.
+     * However, it gets a little more complex is manifest preexists.
+     * So we first try to see if a manifest exists, and otherwise create a new one.
+     * 
+     * <b>BEWARE</b> this method should be invoked after file copy from target/classes and so on.
+     * Otherwise this manifest manipulation will be useless.
+     * 
+     * @see https://github.com/quarkusio/quarkus/issues/1443
+     * @param runnerZipFs
+     * @param classPath
+     * @throws IOException
+     */
+    private void generateManifest(FileSystem runnerZipFs, final String classPath) throws IOException {
+        final Path manifestPath = runnerZipFs.getPath("META-INF", "MANIFEST.MF");
+        final Manifest manifest = new Manifest();
+        if (Files.exists(manifestPath)) {
+            log.warn(
+                    "Seems like you have a pre-existing MANIFEST.MF. Since quarkus generates an executable JAR, some entries will be added.");
+            try (InputStream is = Files.newInputStream(manifestPath)) {
+                manifest.read(is);
+            }
+        }
+        Attributes attributes = manifest.getMainAttributes();
+        attributes.put(Attributes.Name.MANIFEST_VERSION, "1.0");
+        if (attributes.containsKey(Attributes.Name.CLASS_PATH)) {
+            log.warn(
+                    "Your MANIFEST.MF defines a CLASSPATH. We don't yet support merge of CLASSPATHS. So your CLASSPATH will be overwritten by Quarkus one.");
```

Comment:
Something like: "Your MANIFEST.MF already defined a CLASS_PATH entry. Quarkus has overwritten this existing entry."

I usually don't make promises about "potential future support of something" in log messages. Usually, you just forgot about them and they end up being stale or false promises.